src/platform/forms-system/src/js/utilities/data/reduceErrors.js

```diff
@@ -0,0 +1,156 @@
+ * @property {string} pageKey - the page within the chapter the element is
+ *   associated with
+ */
+/**
+ * @typedef Form~rawErrors - list of raw errors that are output from jsonschema
+ *   validator
+ * @type {object}
+ * @property {string} property
+ * @property {string} message
+ * @property {string} name
+ * @property {string|number} argument - element name or index in array
+ * @property {object} schema - ignored
+ * @property {string} stack - error message used in processing
+ * @property {string[]} __errors - may contain multiple error messages for one
+ *   element
+ *
+ * There are four types of hardcoded validation error messages:
+ *
+ * min/max: {
+    property: 'instance.someProperty',
+    message: 'does not meet minimum length of 1',
+    name: 'minItems',
+    argument: 1,
+    schema: {...},
+    stack: 'instance.someProperty does not meet minimum length of 1'
+  }
+  * required: {
+    property: 'instance',
+    message: 'requires property "someProperty"',
+    name: 'required',
+    argument: 'someProperty',
+    schema: {...},
+    stack: 'instance requires property "someProperty"'
+  }
+  * array: {
+    property: 'instance.newDisabilities[0]',
+    message: 'requires property "cause"',
+    name: 'required',
+    argument: 'cause',
+    schema: {...},
+    stack: 'instance.newDisabilities[0] requires property "cause"'
+  }
+  * non-empty "__errors" array: {
+    __errors: ['error message']
+  }
+*/
+/**
+ * @typedef Form~formErrors
+ * @type {object}
+ * @property {Form~rawErrors}
+ * @property {Form~errors}
+ */
+/**
+ * Process rawErrors from jsonschema validator into a more friendly list
+ * @param {Form~formErrors} errors
+ * @param {object[]} pageList - list of all form pages from `form.pages`
+ */
+export const reduceErrors = (errors, pageList) =>
+  errors.reduce((acc, error) => {
+    const findErrors = (key, err) => {
+      if (typeof err === 'object') {
+        if (err.message) {
+          const property = err.property.replace(/instance\.?/, '') || '';
+          const instance = property.replace(/(\[\d+\])/, '');
+          const name = err.name === 'required' ? err.argument : instance;
+          /*
+           * There may be multiple errors for the same property, e.g.
+           * `contestedIssues` in form 996. It can have a custom message
+           * "Please select a contested issue" and a minItem error
+           * "Does not meet minimum length of 1"; there's no need to confuse
+           * anyone and show both
+          */
+          if (!errorExists(acc, name)) {
+            const { chapterKey = '', pageKey = '' } = getPropertyInfo(
+              pageList,
+              err.argument,
+              err.property.startsWith('instance.') ? instance : '',
+            );
+            acc.push({
+              name,
+              // property may be `array[0]`; we need to extract out the `[0]`
+              index: property.match(/\[(\d+)\]/)?.[1] || null,
+              message: err.stack,
```

Comment:
Why use `err.stack` instead of `err.message`?

Reply:
`err.stack` has a more complete message, which makes it easier to manipulate (later, once we process the message for display to the user)

For example:

```js
message: 'requires property "cause"',
stack: 'instance.newDisabilities[0] requires property "cause"'
```

I guess it would make more sense to use `err.message` since the latest error object doesn't include a `stack` value

Reply:
Actually, the `stack` does contain important info that will need to be used once we process the message for the user to see.

```js
message: 'requires property "city"',
stack: 'instance.address requires property "city"',
```

the `address` within the `stack` will help guide the user to the appropriate page section